Makefile

```diff
@@ -1,5 +1,5 @@
-# Specifies the targets that should ALWAYS have their recipies run
-.PHONY: help clean clean-pyc clean-build clean-assets writeversion lint test test-all coverage docs release i18n-crowdin-upload i18n-crowdin-download
+# List most target names as 'PHONY' to prevent Make from thinking it will be creating a file of the same name
+.PHONY: help clean clean-assets clean-build clean-pyc clean-docs lint test test-all assets coverage docs release test-namespaced-packages staticdeps staticdeps-cext writeversion buildconfig pex i18n-extract-frontend i18n-extract-backend i18n-extract i18n-django-compilemessages i18n-upload i18n-download i18n-regenerate-fonts i18n-stats i18n-install-font docker-clean docker-whl docker-deb docker-deb-test docker-windows docker-demoserver docker-devserver
```

Comment:
@benjaoming are we using any of our make targets to actually create files of the same name, except maybe `dist`?

Reply:
@indirectlylit yes, that's the case, so failing to mark a command as ".PHONY" would not likely have any immediate consequences, except in the case of `make dist` because of the `dist` dir.

We _could_ use this pattern of adding `.PHONY` right next to each command... https://gist.github.com/ajmazurie/d8edac58f6774ed4b8cc7dc9c0c81a90

Reply:
yes that looks easier to maintain

Reply:
This discussion `make`s me think that we, in general, overload the Makefile.